### PR TITLE
Keyhint fix

### LINKED ISSUE #1507: Key hint widget is shown when using hints
When using `f` and starting to type a hint, an empty key hint widget is shown (and never hidden again when the hint is completed).

cc @rcorre

### PR DESCRIPTION
**Pull request by rcorre**
Resolve #1507 by preventing the dialog from showing if there are no matches.
The second commit is just for cleanup and replaces an oft-repeated special key check with a utility function.

### CODE REVIEW

File: tests/unit/utils/test_utils.py [hunk 1]
```diff
@@ -984,3 +984,15 @@ def test_get_fake_clipboard(self, selection):
     def test_supports_selection(self, clipboard_mock, selection):
         clipboard_mock.supportsSelection.return_value = selection
         assert utils.supports_selection() == selection
+
```
Comment (The-Compiler) [inline tests/unit/utils/test_utils.py]: 2 blank lines here (between functions)

File: tests/unit/utils/test_utils.py [hunk 2]
```diff
@@ -984,3 +984,15 @@ def test_get_fake_clipboard(self, selection):
...
+@pytest.mark.parametrize('keystr, expected', [
+    ('<Control-x>', True),
+    ('<Meta-x>', True),
+    ('<Ctrl-Alt-y>', True),
+    ('x', False),
+    ('X', False),
+    ('<Escape>', True),
+    ('foobar', False),
```
Comment (The-Compiler) [inline tests/unit/utils/test_utils.py]: Please add a few tests for corner cases like `<<` or `<foo` or `foo>` too.

File: qutebrowser/misc/keyhintwidget.py [hunk 3]
```diff
@@ -94,28 +94,31 @@ def update_keyhint(self, modename, prefix):
             self.hide()
             return
 
+        keyconf = objreg.get('key-config')
+        bindings = [(k, v) for (k, v)
+                    in keyconf.get_bindings_for(modename).items()
+                    if k.startswith(prefix) and not utils.is_special_key(k)]
+
+        if not bindings:
+            return
```
Comment (The-Compiler) [inline qutebrowser/misc/keyhintwidget.py]: Can you add a unit test for this case?

**Comment by The-Compiler** [comment]
Thanks!
